File: ios/Plugin/ZebraPrinterPlugin.swift
import Foundation
import Capacitor
import SwiftSocket

@objc(ZebraPrinterPlugin)
public class ZebraPrinterPlugin: CAPPlugin {

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": value
        ])
    }

    @objc func print(_ call: CAPPluginCall) {
        let ip = call.getString("ip") ?? ""
        let port = call.getInt("port") ?? 0
        let zpl = call.getString("zpl") ?? ""

        let client:TCPClient = TCPClient(address: ip, port: Int32(port))
        switch client.connect(timeout: 30){
            case .success:
                switch client.send(string: zpl) {
                    case .success:
                        guard let data = client.read(1024*10)
                        else {
                            call.resolve()
                            return
                        }
                    case .failure(let error):
                        call.reject("Something went wrong")

                }
            case .failure(let error):
                call.reject("Something went wrong")
        }
    }
}

File: android/src/main/java/com/zebra/plugins/printer/ZebraPrinterPlugin.java
package com.zebra.plugins.printer;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import java.io.DataOutputStream;
import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.Socket;
import java.net.SocketAddress;

@CapacitorPlugin(name = "ZebraPrinter")
public class ZebraPrinterPlugin extends Plugin {

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }

    @PluginMethod
    public void print(PluginCall call) {
        String ip = call.getString("ip");
        int port = call.getInt("port");
        String zpl = call.getString("zpl");

        System.out.println("start");
        System.out.println("------------------------------------------------------------");

        Boolean error = false;
        try {
            Socket clientSocket = null;
            DataOutputStream outToServer = null;
            try {
                System.out.println("Printing now...");
                clientSocket = new Socket();
                SocketAddress socketAddress = new InetSocketAddress(ip, port);
                clientSocket.connect(socketAddress, 30000);
                outToServer = new DataOutputStream(clientSocket.getOutputStream());
                outToServer.writeBytes(zpl);
                System.out.println("send the following string: " + zpl);
            } catch (Exception e) {
                System.out.println("Connection error");
                error = true;
            } finally {
                if (clientSocket != null) {
                    outToServer.close();
                    clientSocket.close();
                }
            }
        } catch (IOException e1) {
            System.out.println("Cannot print label on this printer : " + ip + ":" + port);
            System.out.println(e1.getMessage());
            call.reject(e1.getMessage());
            error = true;
        }
        System.out.println("------------------------------------------------------------");
        System.out.println("end");

        JSObject ret = new JSObject();
        if (error) {
            ret.put("success", false);
            ret.put("message", "Something went wrong");
        } else {
            ret.put("success", true);
            ret.put("message", "Successfully sent to printer");
        }
        call.resolve(ret);
    }
}
